assets/js/src/integrations/classic.js

```diff
@@ -0,0 +1,132 @@
+			tracker.eventHandler( eventName )( eventData );
+		}
+	} );
+
+	/**
+	 * Track the custom add to cart event dispatched by WooCommerce Core
+	 *
+	 * @param {Event} e - The event object
+	 * @param {Object} fragments - An object containing fragments of the updated cart.
+	 * @param {string} cartHash - A string representing the hash of the cart after the update.
+	 * @param {HTMLElement[]} button - An array of HTML elements representing the add to cart button.
+	 */
+	document.body.onadded_to_cart = ( e, fragments, cartHash, button ) => {
+		tracker.eventHandler( 'add_to_cart' )( {
+			product: getProductFromID(
+				parseInt( button[ 0 ].dataset.product_id )
+			),
+		} );
+	};
+
+	/**
+	 * Attaches click event listeners to all remove from cart links
+	 */
+	const removeFromCartListener = () => {
+		document
+			.querySelectorAll(
+				'.woocommerce-cart-form .woocommerce-cart-form__cart-item .remove[data-product_id]'
+			)
+			.forEach( ( item ) =>
+				item.addEventListener( 'click', removeFromCartHandler )
+			);
+	};
+
+	/**
+	 * Handle remove from cart events
+	 *
+	 * @param {HTMLElement|Object} element - The HTML element clicked on to trigger this event
+	 */
+	function removeFromCartHandler( element ) {
+		tracker.eventHandler( 'remove_from_cart' )( {
+			product: getProductFromID(
+				parseInt( element.target.dataset.product_id )
+			),
+		} );
+	}
+
+	// Attach event listeners on initial page load and when the cart div is updated
+	removeFromCartListener();
+	document.body.onupdated_wc_div = () => removeFromCartListener();
+
+	// Trigger the handler when an item is removed from the mini-cart and WooCommerce dispatches the `removed_from_cart` event.
+	document.body.onremoved_from_cart = (
+		event,
+		fragments,
+		/* eslint-disable-next-line camelcase */
+		cart_hash,
+		button
+	) => removeFromCartHandler( { target: button[ 0 ] } );
+
+	/**
+	 * Attaches click event listeners to non-block product listings that sends a
+	 * `select_content` event if the target link takes the user to the product page.
+	 */
+	document
+		.querySelectorAll( '.products .product:not(.wp-block-post)' )
+		?.forEach( ( item ) => {
+			// Get the Product ID from a child node containing the relevant attribute
+			const productId = item
+				.querySelector( 'a[data-product_id]' )
+				?.getAttribute( 'data-product_id' );
+
+			if ( ! productId ) {
+				return;
+			}
+
+			item.addEventListener( 'click', ( event ) => {
+				// Return early if the user has clicked on anything other
+				// than a product link or an Add to cart button.
+				const targetLink = event.target.closest(
+					'.woocommerce-loop-product__link'
+				);
+
+				const isButton = event.target.classList.contains( 'button' );
+
+				const isAddToCartButton =
+					event.target.classList.contains( 'add_to_cart_button' ) &&
+					! event.target.classList.contains(
+						'product_type_variable'
+					);
+
+				if ( ! targetLink && ( ! isButton || isAddToCartButton ) ) {
+					return;
+				}
```

Comment:
:nail_care: 
I got a gut feeling that this selector may be hard to keep in sync to match with WC core/theme changes.

Should we fire the `add_to_cart` event for non-ajax buttons as well?
From reading WC Core code, I think we can disable `ajax_add_to_cart` globally or for a product, then end up with a 
`<a class="button add_to_cart_button">Add to cart</a>`.
If we want to fire for those, then the `!isButton` condition should be removed. Otherwise, we could use `.ajax_add_to_cart` to precisely select only those add-to-cart-like buttons that actually add to cart, guarding against `.product_type_variable` to any other.

Are you sure we will not have some other `.button` in the product card that does not actually select the product, but does some other thing? Like a link to a promotion, group, brand site, etc?
Maybe we can identify "Add to cart"-likeness by having `data-product_id` attribute :) 
So we would do 
```suggestion
				const isProductLink =
					// Regular product link.
					event.target.closest( '.woocommerce-loop-product__link' ) ||
					// Add-to-cart-like button.
					event.target.hasAttribute( 'data-product_id' );

				const isAddToCartButton =
					event.target.classList.contains( 'add_to_cart_button' ) &&
					! event.target.classList.contains(
						'product_type_variable'
					);

				if ( ! isProductLink || isAddToCartButton ) {
					return;
				}
```


Also, all this confusion looks like an encouragement to champion a PR to WC Core to add the product_id datar attribute to the product card ;)

Reply:
When clicking on non-ajax add to cart button the event will be sent when the page [loads and the confirmation message is displayed](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/94cc036583bc03f5b9ef6dd829401ab9a4207225/includes/class-wc-abstract-google-analytics-js.php#L80) so I don't think we need additional targeting there.

I've added a check like you suggested so that the event will only send if the button includes the `data-product_id` attribute to reduce the chance of issues with other extensions in 94cc036.